department-of-veterans-affairs/caseflow
# Show errors in queue flow modal

Resolves #14216

### Description
Ensures that modals do no close if there is an error to show. Also suppresses success and error alerts on the case details page if there is a modal open.

### Acceptance Criteria
- [x] Assign modal does not close if the submit was not successful
- [x] Assign modal does shows an error if the submit was not successful
- [x] Case details page does not show an error behind the modal if the submit was not successful

### Testing Plan
1. Log in as BVAAABSHIRE and go to may case with a judge assign task
1. Add `return invalid_type_error` before `queue_for_role = QueueForRole...` in [`JudgeAssignTasksController#create`](https://github.com/department-of-veterans-affairs/caseflow/blob/89d0a65d8e55a967d8e3e4e77eb99c72b408ba5a/app/controllers/judge_assign_tasks_controller.rb#L4)
1. Attempt to assign the case to an attorney
1. Ensure an error is shown in the modal rather than continuing forward to queue
1. Ensure no error is shown _behind_ the modal 

### User Facing Changes
![Screen Shot 2020-06-01 at 9 50 45 AM](https://user-images.githubusercontent.com/45575454/83416128-341bc300-a3ee-11ea-88fc-64a2fecd648c.png)
